addons/website_slides/static/src/js/slides_course_slides_list.js

```diff
@@ -1,251 +1,116 @@
+        });
+    },
+
+    /**
+     * Bind the sortable jQuery widget to both
+     * - course sections
+     * - course slides
+     *
+     * @private
+     */
+    _bindSortable: function () {
+        this.$('ul.o_wslides_channel').sortable({
+            handle: '.fa-arrows',
+            stop: this._reorderCategories.bind(this)
+        });
+
+        this.$('.o_wslides_channel ul').sortable({
+            handle: '.fa-arrows',
+            connectWith: '.section-draggable ul',
+            stop: this._reorderSlides.bind(this)
+        });
+    },
+
+    _reorderSlides: function (){
+        this._rpc({
+            route: "/slides/resequence_slides",
+            params: {
+                slides_data: this._getSlides()
             }
-        },
-        _reorderSlides: function(){
-            var self = this;
-            // In case the slide was transfered to another section
-            if (self.draggedElement.hasClass('slide-draggable')){
-                self.draggedElement.attr('category_id', parseInt(self.dropTarget.attr('category_id')))
+        });
+    },
+
+    _reorderCategories: function (){
+        var self = this;
+        self._rpc({
+            route: '/web/dataset/resequence',
+            params: {
+                model: "slide.category",
+                ids: self._getCategories()
             }
-            self.slides = [];
-            self._getSlides();
-            self._rpc({
-                route: "/slides/resequence_slides",
-                params: {
-                    slides_data: self.slides
-                }
-            }).then(function(){
-            });
-        },
-        _rebindUploadButton: function(categoryID){
-            var self = this;
-            this.$('.oe_slide_js_upload[data-category-id='+categoryID+']').click(function(ev){
-                ev.preventDefault();
-                var data = $(ev.currentTarget).data();
-                var dialog = new SlideUpload.SlideUploadDialog(self, data);
-                dialog.appendTo(document.body);
-                dialog.open();
-            })
-        }
-    })
-
-    sAnimations.registry.websiteSlidesCourseSlidesList = Widget.extend({
-        selector: '.oe_js_course_slides_list',
-        xmlDependencies: ['/website_slides/static/src/xml/website_slides_upload.xml'],
-        init: function (el){
-            this._super.apply(this, arguments);
-        },
-        start: function (){
-            this._super.apply(this, arguments);
-            var list = new List(this);
-            list.appendTo(".oe_js_course_slides_list");
-        }
-    });
-});
\ No newline at end of file
+        }).then(function (){
+            self._resetCategoriesIndex();
+        });
+    },
+
+    /**
+     * Returns a slides dict in the form:
+     * {slide_id: {'sequence': slide_sequence, 'category_id': slide.category_id.id}}
+     *
+     * @private
+     */
+    _getSlides: function (){
```

Comment:
Side comment: we could imagine ordering methods by name so that "that team" will not scream too much.

Reply:
"That team" should be significantly happier now.